### PR TITLE
Get it to build on Haiku + additional cleanups

### PR DESCRIPTION
**Pull request by BachToTheFuture**
This is so that I could write a recipe instead of having to use patches :)
Updates:
* converted README to markdown
* added rdef file
* added Makefile
* minor code changes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+#%{
+# @src->@ 
+
+#	Specify the source files to use. Full paths or paths relative to the 
+#	Makefile can be included. All files, regardless of directory, will have
+#	their object files created in the common object directory. Note that this
+#	means this Makefile will not work correctly if two source files with the
+#	same name (source.c or source.cpp) are included from different directories.
+#	Also note that spaces in folder names do not work well with this Makefile.
+SRCS =  src/CWindow.cpp \
```
Comment (humdingerb) [inline Makefile]: Cosmetics: if you have one file per line, it looks nicer like this:
```
SRCS = \
		src/CWindow.cpp \
		src/CView.cpp \
		src/CSearch.cpp \
		src/CMain.cpp \
		src/CBoard.cpp \
		src/CApp.cpp
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+#		you can simply specify XXX for the library. (e.g. the entry for
+#		"libtracker.so" would be "tracker")
+#
+#	-	for GCC-independent linking of standard C++ libraries, you can use
+#		$(STDCPPLIBS) instead of the raw "stdc++[.r4] [supc++]" library names.
+#
+#	- 	if your library does not follow the standard library naming scheme,
+#		you need to specify the path to the library and it's name.
+#		(e.g. for mylib.a, specify "mylib.a" or "path/mylib.a")
+LIBS =  $(STDCPPLIBS) be media tracker game
```
Comment (humdingerb) [inline Makefile]: Only "be" is needed by the app.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+#	or libXXX.a naming scheme. You can specify full paths or paths relative
+#	to the Makefile. The paths included are not parsed recursively, so
+#	include all of the paths where libraries must be found. Directories where
+#	source files were specified are	automatically included.
+LIBPATHS = 
+
+#	Additional paths to look for system headers. These use the form
+#	"#include <header>". Directories that contain the files in SRCS are
+#	NOT auto-included here.
+SYSTEM_INCLUDE_PATHS =  /boot/system/develop/headers/be \
```
Comment (humdingerb) [inline Makefile]: These are all default and can be removed.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+						/boot/system/develop/headers/cpp \
+						/boot/system/develop/headers/posix
+
+#	Additional paths paths to look for local headers. These use the form
+#	#include "header". Directories that contain the files in SRCS are
+#	automatically included.
+LOCAL_INCLUDE_PATHS =  . src
```
Comment (humdingerb) [inline Makefile]: These paths aren't used and can be removed.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+#	Specify the level of optimization that you want. Specify either NONE (O0),
+#	SOME (O1), FULL (O2), or leave blank (for the default optimization level).
+OPTIMIZE := NONE
```
Comment (humdingerb) [inline Makefile]: Compies and runs with the default. Remove "NONE".

**Review by humdingerb — CHANGES_REQUESTED** [review]
While you're at it, can you change the app sig in the RDEF and CApp.h to `application/x-vnd.Connect-4`, please.

**Comment by BachToTheFuture** [comment]
@humdingerb Applied some changes 😄

File: README.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
+# Connect-4 v1.0
```
Comment (humdingerb) [inline README.md]: I'd remove the versioning here. That way it doesn't have to be touched for future versions.

File: README.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+### by Leo Chan, (c)1997
+
+![screenshot](screenshot.png)
+
+### Source information
```
Comment (humdingerb) [inline README.md]: I'd removee everthing in "Source information". It's either outdated or low on real information...

File: README.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+* Description: A simple game of Connect 4
+
+* Notes: The project for this game is in the Project directory.
+
+
+
+Please mail any comments to leochan@gpu.srv.ualberta.ca
+
+This is the game of Connect-4.  The game is played by players taking turns dropping pieces down a column.  The first player to get four consecutive pieces in a row, column, or diagonal wins the game.
```
Comment (humdingerb) [inline README.md:8]: As that email address is almost guaranteed to be dead, I'd just keep the screenshot and this paragraph.

File: src/Connect4.rdef [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+	$"202020102A2A2A102020201020202003"
+	$"10201010102A10101020101010201003"
+	$"10101010101010101010101010101003"
+	$"1020101010201010102A101010201003"
+	$"20202010202020102A2A2A1020202003"
+	$"1020101010201010102A101010201003"
+	$"03030303030303030303030303030303"
+};
+
+resource app_signature "application/x-vnd.Connect-4";
```
Comment (humdingerb) [inline src/Connect4.rdef]: Please move everything from line 58ff up to the top of the file.

**Review by humdingerb — CHANGES_REQUESTED** [review]

File: src/Connect4.rdef (hunk 9 above)
Comment (BachToTheFuture) [reply src/Connect4.rdef]: @humdingerb So I just put everything from line 58 and after to the beginning of the file like this?

Comment (humdingerb) [reply src/Connect4.rdef]: Exactly like this. Thanks.

**Review by humdingerb — APPROVED** [review]

**Comment by humdingerb** [comment]
Thanks.

**Pull request merged**